src/workflows-app/SubmitWorkflow.js

```diff
@@ -86,48 +94,79 @@ export const SubmitWorkflow = wrapWorkflowsPage({ name: 'SubmitWorkflow' })(
       load();
     });
 
+    useEffect(() => {
+      const refresh = async () => {
+        await refreshApps(true);
+      };
+
+      if (!currentApp || getIsAppBusy(currentApp)) {
+        refresh();
+      }
+    }, [currentApp, refreshApps]);
+
+    const createWorkflowsApp = Utils.withBusyState(setCreating, async () => {
+      try {
+        setCreating(true);
+        await Ajax(signal).Apps.createAppV2(Utils.generateAppName(), workspace.workspace.workspaceId, appToolLabels.CROMWELL);
+        await Ajax(signal).Metrics.captureEvent(Events.applicationCreate, {
+          app: appTools.CROMWELL.label,
+          ...extractWorkspaceDetails(workspace),
+        });
+        await refreshApps(true);
+      } catch (error) {
+        reportError('Cloud Environment Error', error);
+      } finally {
+        setCreating(false);
```

Comment:
Is the `finally` block needed? I assumed that `withBusyState` would set the `setCreating` to `false` when it has finished running the function passed to it no? 🤔

Reply:
it is! this was actually a key ingredient to making this behave correctly.

Reply:
Ah I see. Interesting. Thank you for clarifying!